nextcord/application_command.py

```diff
@@ -0,0 +1,865 @@
+        """Optional[:class:`Guild`]: Returns the :class:`Guild` associated with this Response, if any."""
+        return self._state._get_guild(self.guild_id)
+
+    @property
+    def signature(self) -> Tuple[str, int, Optional[int]]:
+        """Returns a simple high level signature of the command. No commands registered in the bot at the same time
+        should have identical signatures.
+
+        Returns
+        -------
+        name: :class:`str`
+            Name of the Application Command.
+        type: :class:`int`
+            Discord's integer value of the Application Command type
+        guild_id: Optional[:class:`int`]
+            The Guild ID associated with the Application Command. If None, it's a global command.
+        """
+        return self.name, self.type.value, self.guild_id
+
+
+class ApplicationCommandResponseOptionChoice:
+    """Represents a single choice in a list of options.
+
+    Attributes
+    ----------
+    name: :class:`str`
+        Name of the choice, this is what users see in Discord.
+    value: Union[:class:`str`, :class:`int`, :class:`float`]
+        Value of the choice, this is what Discord sends back to us.
+    """
+
+    def __init__(self, payload: Optional[dict] = None):
+        if not payload:
+            payload = {}
+        self.name: str = payload.get('name')
+        self.value: Union[str, int, float] = payload.get('value')
+
+
+class ApplicationCommandResponseOption:
+    """Represents an argument/parameter/option or subcommand of an Application Command.
+
+    Attributes
+    ----------
+    type: :class:`ApplicationCommandOptionType`
+        Enum corresponding to the Application Command Option type. (subcommand, string, integer, etc.)
+    name: :class:`str`
+        Name of the option or subcommand.
+    description: :class:`str`
+        Description of the option or subcommand.
+    required: :class:`bool`
+        If this option is required for users or not.
+    """
+
+    def __init__(self, payload: dict):
+        self.type = ApplicationCommandOptionType(int(payload["type"]))
+        self.name: str = payload['name']
+        self.description: str = payload['description']
+        self.required: Optional[bool] = payload.get('required')
+        self.choices: List[ApplicationCommandResponseOptionChoice] = self._create_choices(payload.get('choices', []))
+        self.options: List[ApplicationCommandResponseOption] = self._create_options(payload.get('options', []))
+
+    @staticmethod
+    def _create_choices(data: List[dict]) -> List[ApplicationCommandResponseOptionChoice]:
+        return [ApplicationCommandResponseOptionChoice(raw_choice) for raw_choice in data]
+
+    @staticmethod
+    def _create_options(data: List[dict]) -> List[ApplicationCommandResponseOption]:
+        return [ApplicationCommandResponseOption(raw_option) for raw_option in data]
+
+
+class ClientCog:
+    # TODO: I get it's a terrible name, I just don't want it to duplicate current Cog right now.
```

Comment:
bapboop wasnt this ``CommandCog`` before so is this TODO needed

Reply:
I'd argue that it's still a terrible name, but if others are fine with it then I can remove that.